File: algorithms/sorting/bubble_sort/bubble_sort.js
function bubbleSort(arr) {
    let unsortedUntilIndex = arr.length - 1;
    let sorted = false;
    while (!sorted) {
        sorted = true;
        for (let i = 0; i < unsortedUntilIndex; i++) {
            if (arr[i] > arr[i + 1]) {
                let temp = arr[i + 1];
                arr[i + 1] = arr[i];
                arr[i] = temp;
                sorted = false;
            }
        }
        unsortedUntilIndex--;
    }
    return arr;
}

File: algorithms/sorting/bubble_sort/bubble_sort.py
def bubble_sort(arr):
    unsorted_until_index = len(arr) - 1
    sorted = False
    while not sorted:
        sorted = True
        for i in range(unsorted_until_index):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                sorted = False
        unsorted_until_index -= 1
    return arr
